demos/add_exporter/init.js

```diff
@@ -0,0 +1,27 @@
+function updateExporters(exporters) {
+    for (const exporter of exporters) {
+        exporter.isDefault = false;
+    }
+    exporters.push({
+        extension: "csv",
+        label: "My fancy CSV",
+        showDeckList: true,
+        showIncludeScheduling: false,
+        showIncludeDeckConfigs: false,
+        showIncludeMedia: false,
+        showIncludeTags: false,
+        showIncludeHtml: false,
+        showLegacySupport: false,
+        showIncludeDeck: false,
+        showIncludeNotetype: false,
+        showIncludeGuid: false,
```

Comment:
In the deck options screen we explored extending the UI with Svelte, and with raw HTML/JS. The former required some hacks at bundling time, and I'm not sure they'll continue to work in the upgrade to Svelte 5.

The main argument for doing it is so add-on authors can make use of our components, to better match look&feel. But it means tighter coupling to our code, and makes things more likely to break. I find myself wondering whether the latter might be the better choice.

Alternatively we could go down the param route, like you suggested. It would probably work for the common case, but a drawback would be that add-on authors would be limited to the GUI elements we expose.

I'm not entirely sure what the best path forward is - it might be best if this were broken into a separate discussion and we got other people like @abdnh and @glutanimate participating as well.

Reply:
Moved to https://github.com/ankitects/anki/issues/3187.